viewing_party/party.py

```diff
@@ -0,0 +1,243 @@
+        rating_sum += movie_dict["rating"]
+
+    rating_avg = rating_sum / (len(user_data["watched"]))
+    return rating_avg
+
+
+#wave 2, part 2
+def get_most_watched_genre(user_data):
+    """Decides most popular genre from watched list. Returns string."""
+
+    genre_list = []
+    fantasy_count = 0
+    action_count = 0
+    intrigue_count = 0
+
+    if len(user_data["watched"]) == 0:
+        return None
+
+    for movie_dict in user_data["watched"]:
+        genre_list.append(movie_dict["genre"])
+
+    fantasy_count = genre_list.count("Fantasy")
+    action_count = genre_list.count("Action")
+    intrigue_count = genre_list.count("Intrigue")
+
+    if (fantasy_count > action_count) and (fantasy_count > intrigue_count):
+        return "Fantasy"
+    elif (action_count > fantasy_count) and (action_count > intrigue_count):
+        return "Action"
+    else: 
+        return "Intrigue"
+
+
+
+
+
+
+
+
+################ WAVE 3 ##############
+
+#wave 3, part 1
+def get_unique_watched(user_data):
+    """Takes dict, returns list of dicts that only user has seen."""
+    
+    only_user_watched = []
+    user_watched_set = set()
+    friends_watched_set = set()
+    only_user_watched_set = set()
+
+    for movie_user_watched in user_data["watched"]:
+        user_watched_set.add(movie_user_watched["title"])
+        for friends_watched_lists in user_data["friends"]:
+            for movies_friends_watched_list in friends_watched_lists.values():
+                for indiv_dict in movies_friends_watched_list:
+                    friends_watched_set.add(indiv_dict["title"])
+
+    only_user_watched_set = user_watched_set.difference(friends_watched_set)
+
+    for item in only_user_watched_set:
+        only_user_watched.append({"title" : item})
+
+    return only_user_watched
+                    
+    
+
+#wave 3, part 2
+def get_friends_unique_watched(user_data):
+    """Takes dict, returns list of dicts that only friends have seen."""
+
+    user_watched_set = set()
+    friends_watched_set = set()
+    only_friends_watched_set = set()
+    only_friends_watched = []
+
+    for movie_user_watched in user_data["watched"]:
+        user_watched_set.add(movie_user_watched["title"])
+        
+    for friends_watched_lists in user_data["friends"]:
+        for movies_friends_watched_list in friends_watched_lists.values():
+            for indiv_dict in movies_friends_watched_list:
+                friends_watched_set.add(indiv_dict["title"])
+
+    only_friends_watched_set = friends_watched_set.difference(user_watched_set)
+
+    for item in only_friends_watched_set:
+        only_friends_watched.append({"title" : item})
+
+    return only_friends_watched
+
+
+
+
+################ WAVE 4 ##############
+
+def get_available_recs(user_data):
+    """Takes dict, returns list of movie recommendations."""
+
+    friends_watched_list = get_friends_unique_watched(user_data)
+    friends_watched_list_titles = []
```

Comment:
Consider using a set rather than a list here since you're using this to store the result titles for lookup purposes. On line 178, the `in` lookup for lists is O(n), but the `in` lookup for sets is O(1).